bundles/org.openhab.binding.argoclima/README.md

```diff
@@ -0,0 +1,423 @@
+| localDeviceIP             | text           | Local IP address of the device for matching intercepted requests (may be different from **hostname**, if behind NAT). Used in ```REMOTE_API_*``` modes.                            | N/A              | no       | yes      |
+| deviceCpuId               | text           | CPU ID of the device. Optional, value is used for detecting device update in proxy mode.  Used in ```REMOTE_API_*``` modes.                                                        | N/A              | no       | yes      |
+| useLocalConnection        | boolean        | Whether the binding is permitted to talk to the device directly.                                                                                                                   | yes              | no       | yes      |
+| refreshInterval           | integer        | Interval the device is polled in (in sec.). Used if ```useLocalConnection== true```                                                                                                | 30               | no       | yes      |
+| stubServerPort            | integer        | Stub server listen port  Used in ```REMOTE_API_*``` modes.                                                                                                                         | 8239             | no       | yes      |
+| stubServerListenAddresses | text(multiple) | List of interfaces the stub server will listen on  Used in ```REMOTE_API_*``` modes.                                                                                               | 0.0.0.0          | no       | yes      |
+| oemServerAddress          | text           | The Argo server's IP or hostname, used for pass-through. Used in ```REMOTE_API_PROXY``` mode                                                                                       | 31.14.128.210    | no       | yes      |
+| oemServerPort             | integer        | The Argo server's port. Used in ```REMOTE_API_PROXY``` mode                                                                                                                        | 80               | no       | yes      |
+| showCleartextPasswords    | boolean        | Show passwords sent by device in clear text  Used in ```REMOTE_API_*``` modes.                                                                                                     | false            | no       | yes      |
+| matchAnyIncomingDeviceIp  | boolean        | If enabled, will accept any Argo message as matching this Thing (instead of requiring an exact-match by ```hostname``` or ```localDeviceIP```).  Used in ```REMOTE_API_*``` modes. | no               | no       | yes      |
+
+### General device configuration (dynamic)
```

Comment:
They technically only are in effect when in one of the schedule modes (so these would be co-depentent channels). 
TBH though, I found it easier of a paradigm to have them configured jointly in one place (with nice MainUI-assisted UX) than as a bunch of channels.
My rationale for it is mostly how the remote control for this device operates. It allows you to _configure_ Schedules 1|2|3 (via entering a submenu of sorts) , and then you get to _select_ one of the modes via push of a button (which sends the device the complete collection of all relevant parameters).

So... I thought it's closer to the original experience, and this being openHab... if someone wants ultimate flexibility... rules are in any case superior.

---
All that said, this is my thinking. I'm happy to be persuaded otherwise!

Reply:
Moddeling it as channels would also give about the same UX experience, is more friendly to text based configurations and would make it possibble to alter the schedule from rules.

I don't say it is wrong (i'm no maintainer), but i'm not aware of other bindings using the configuration like this.

Reply:
> Moddeling it as channels would also give about the same UX experience

Hmm... Perhaps I'm missing something very obvious then! 
The setting I found most awkward to work with is "schedule enabled days" (basically an `EnumSet<Weekday>`.
When I model it as a configuration, I can get to something like:
![image](https://github.com/openhab/openhab-addons/assets/2566147/bbb76f10-c432-4ae6-8b0f-30ca4961705d)
... whereas when I make it a channel, I thought I had to resort to a... `String` with no baked-in default widgets to help support? Am I mistaken? 

----
To be fair, current UX I have in main UI is not that rich (I had to resign from the "nicer" properties as every entry to thing page was counted as an "update" due to some glitch with MainUI itslf (I think) I wasn't able to narrow down quick...
.. yet the very existence of such widgets pushed me into thinking this is well within the working paradigms (I have Z-Wave and HomeMatic devices that support their built-in schedule, and these are also accessed through config, not through channels).


All in all, I **stand by my offer of remodeling it** if you have a strong preference either way. At the same time, it was quite a deliberate choice (I had it modeled both ways, and decided this way is **way** easier to use, without sacrificing much (or anything, really, as 'dynamic schedule' makes little to no sense, to be performed using these static parameters IMHO).
If you'd be willing, I'd request you to actually try installing the binding, to get a hang of how it feels and let me know if it is awkward indeed.

Reply:
I'm not saying you should do one or the other. For now it was mainly to understand the way it is modelled. My personal preference/feeling is to use channels, there are widgets that could make the UI even more elegant. But i can also understand your view. Maybe a maintainer can comment.

Reply:
BTW:
> would make it possibble to alter the schedule from rules.

AFAIR, you *can* alter configuration from rules. At least I rememeber doing things like:
```python
thing.configuration.put("HMP_0_BOOST_TIME_PERIOD", "5 min")
```
in my jsr223/python rules (don't have a ruleDSL example, as am not using these myself).

Reply:
### Summary
**tl;dr;** Current modeling of "_Schedule timer_" is a result a conscious pro/con analysis. I will of course consider a remodel if this is considered a **must-change** for the PR to be accepted, but strongly prefer to keep as-is.

----
#### Details
My reasoning (for "keep as-is")

1. **Semantics** - this **is** an actual device-side _configuration_. 
    - natively accessed through `⚙️ ->timer` submenu in vendor's APP/remote
    - there's a well-establish precedent (ex. Z-Wave, Homematic bindings) for modelling device's configuration this way (ex. I have a HM thermostat with **this exact** '_feel_'). 
    
2. **Complexity** - schedule settings are NOT standalone _(and would create a concept of interdependent/conditionally-effective channels)_.
   - This is possible to implement but with **considerable** code overhead, IMHO violating Occam's Razor

3. **UX** - modeling as thing config gives an easy-to-use configuration interface within MainUI, w/o the need to maintain/configure channels (at the expense of less flexibility)
   - Yes, it is possible to create same or superior UX through channels, but *not* out of the box
   - With condiguration, the channel-like flexibility can be regained (it is very possible to edit thing config from rules, but see pt. 4)

5. **YAGNI** - Reason I added these settings it is for **full** parity with the remote, though TBH it is only useful for most basic of operations (the schedule limitation the device has is to do `on/off` in the last mode)
   - When fully integrated w/ openHAB, the rules are **way** superior to the schedule settings, giving equivalent featureset and superior configurability
   - The power of the schedule is "configure once, just use as-is" (being able to configure it fixed through a Thing file is an asset, not a liability). Any advanced user will pivot to rules anyway.